### DISCUSSION TITLE
guide/markdown/experiment/

### DISCUSSION THREAD

**Discussion by giscus** [pengzhanbo/vuepress-theme-plume#231] [docs-comment]
# guide/markdown/experiment/

twoslash 为代码块添加支持 TypeScript TwoSlash 支持。 在代码块内提供内联类型提示。 该功能由 shiki 和 @shikijs/twoslash 提供支持， 并整合在 @vuepress-plume/plugin-shikiji 中。 重要 从 vuepress@2.0.0-rc.12 开始，不再需要对 @vuepress...

https://theme-plume.vuejs.press/guide/markdown/experiment/

**Comment by QiHuang02** [comment]
我有个问题，Two Slash 的 `@filename: <filename>` 是可以读取外部 ts 文件来获得内联代码提示吗？还是说得先把 ts 代码写入 md 文件之后在其他 md 文件中用 import 导入捏

**Reply by QiHuang02** [reply]
嗷，可以用Express 来使用自己的定义文件来获得内联提示

**Reply by pengzhanbo** [reply]
你可以把代码块当做是一个执行上下文，`@filename` 将这个代码块拆分为不同的虚拟文件，并可以在一个虚拟文件中引入其它虚拟文件，实现类型解析获取内联代码提示。
代码块内的 `import` 语句，twoslash 同样会去查找对应的包进行类型解析，只要这个包在 `node_modules`中。
